### PR TITLE
Fix failing rq worker in mac os

### PR DESCRIPTION
**Pull request by RajatGoyal**
## What type of PR is this? (check all applicable)
- Bug Fix

## Description
HerokuWorker uses SIGRTMIN which is not supported in MacOSx

## Related Tickets & Documents
https://github.com/getredash/redash/issues/5413

### CODE REVIEW

**Comment by vnnw** [comment]
This patch did fixed the "AttributeError: module 'signal' has no attribute 'SIGRTMIN'" on my local Mac os dev environment.
Thanks for the contribution.
Will anyone in charge review and merge this request please?

File: redash/tasks/worker.py [hunk 1]
```diff
@@ -3,13 +3,21 @@
...
 from rq.timeouts import UnixSignalDeathPenalty, HorseMonitorTimeoutException
 from rq.job import Job as BaseJob, JobStatus
 
-
+# HerokuWorker does not work in OSX https://github.com/getredash/redash/issues/5413
+def get_worker_base_class():
+    if platform == "darwin":
+        return BaseWorker
+    else:
+        return HerokuWorker
```
Comment (arikfr) [inline redash/tasks/worker.py:19]: A more idiomatic way would be:

```python
if platform == "darwin":
    Worker = BaseWorker
else:
    Worker = HerokuWorker
```

And then subclass from `Worker` where you used `get_worker_base_class` before.

**Review by arikfr — COMMENTED** [review]
Thank you for fixing this!

**Comment by konnectr** [comment]
PR  #6222

**Comment by justinclift** [comment]
Merged in https://github.com/getredash/redash/commit/8376e4168448f4b6e9e3d154320c518c10345ab8.

That took some effort. :wink:

**Pull request closed without merging**